src/v/coproc/router.cc

```diff
@@ -384,4 +372,109 @@ router::reader_cfg(model::offset start, model::offset end) {
       _abort_source);
 }
 
+ss::future<> router::process_additions() {
+    return ss::do_with(
+      std::move(_deferred_additions),
```

Comment:
is the purpose here to steal whatever is in `deferred_additions` and leave it empty? if so, i think we should use a something like std::exchange which makes that explicit.

Reply:
yup, agreed

Reply:
this is marked resolved, but i don't see any changes.